chains/evm/calls/events/listener.go

```diff
@@ -30,8 +35,14 @@ func NewListener(client ChainClient) *Listener {
 }
 
 func (l *Listener) FetchDeposits(ctx context.Context, contractAddress common.Address, startBlock *big.Int, endBlock *big.Int) ([]*Deposit, error) {
+	tp := otel.GetTracerProvider()
+	ctx, span := tp.Tracer("relayer-core-tracer").Start(context.Background(), "relayer.core.Listener.FetchDeposits")
```

Comment:
I would prefer have the tracer initialized for the `core` library in some init() library and reused.

Reply:
It is[ initialized in opentelemetry package](https://github.com/ChainSafe/chainbridge-core/pull/346/files#diff-72fd4ae1cd974023e26b0fe861ab81ee641c4aa78d50825309e04d0f9016fe41R39), this one returns it from global vars.

Reply:
Does tp.Tracer("...") create a new trace or you can reuse that tracer across the repo and then Start creates the parent span (which also creates the trace)?

Reply:
The documentation says "The passed name needs to uniquely identify instrumentation code.... If the same name and options are passed multiple times, the same Tracer will be returned"
So basically it uses Params set in the Init function and then you can distinguish them by the namespaces, the problem basically comes more with different UI that are used to display traces, bcs eventually DataDog, Jaeger, etc, parsing this a bit differently.

Reply:
The only thing I want to avoid is having to do GetTracerProvider and then Tracer('some hardcoded string').
Might as well then do the same thing as the logger and keeping a singleton of it somewhere in the trace package (also I think chainbridge-core-tracer might be a btter name for it) and in the functions just do tracer.Start(context, 'functionName').

Having the setup in each function would introduce a lot of bloat and would make the actual logic hard to read.

Reply:
Make sense that it might be a bit cumbersome, but we do not need any custom singletons here as I understand. Anyway I found a builtin function that allowed to replace 2 lines with one and make everything more readable, please check. 
Also DD is not really showing any much difference for an instrumentation name (at least as far as it works for us)   so I just replaced all with relayer-core, and everything relayed to sygma will be relayer-sygma. Will see, we need to leave with the current setup, bcs I am frankly speaking not sure how the end result should look like